pkg/chains/formats/intotoite6/intotoite6.go

```diff
@@ -58,260 +44,16 @@ func (i *InTotoIte6) Wrap() bool {
 }
 
 func (i *InTotoIte6) CreatePayload(obj interface{}) (interface{}, error) {
-	var tr *v1beta1.TaskRun
 	switch v := obj.(type) {
-	case *v1beta1.TaskRun:
-		tr = v
+	case *objects.TaskRunObject:
+		return taskrun.GenerateAttestation(i.builderID, v, i.logger)
```

Comment:
optional (depending on how you feel): you could also define this on the Run objects themselves and do something like

```go
obj.GenerateAttestation(i.builderID)
```

Reply:
Yeah that's a good idea. I think it might be a good idea to have them in same directory as they are now just because of the amount of code reuse between them. Generating attestations doesn't seem to unique to the object. This may change in the future though. Let me know if you agree.